### PR TITLE
Add unit tests generated by Diffblue Deeptest

### PR DESCRIPTION
**Pull request by JohnLBergqvist**

### CODE REVIEW

File: aws-java-sdk-core/pom.xml [hunk 1]
```diff
@@ -101,6 +101,24 @@
...
+      <groupId>org.powermock</groupId>
+      <artifactId>powermock-module-junit4</artifactId>
+      <scope>test</scope>
+      <version>1.6.5</version>
+    </dependency>
+    <dependency>
+      <groupId>com.diffblue</groupId>
+      <artifactId>deeptestutils</artifactId>
+      <scope>test</scope>
+      <version>1.7.1</version>
```
Comment (petersear) [inline aws-java-sdk-core/pom.xml]: deeptestutils is up to 1.8.0 now

File: aws-java-sdk-core/src/test/java/com/amazonaws/http/HttpMethodNameTest.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.amazonaws.http;
+
+import com.amazonaws.http.HttpMethodName;
+import com.diffblue.deeptestutils.Reflector;
```
Comment (petersear) [inline aws-java-sdk-core/src/test/java/com/amazonaws/http/HttpMethodNameTest.java]: No reflection used in tests for this class so could remove.

File: aws-java-sdk-core/src/test/java/com/amazonaws/http/HttpMethodNameTest.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.lang.reflect.Array;
```
Comment (petersear) [inline aws-java-sdk-core/src/test/java/com/amazonaws/http/HttpMethodNameTest.java]: Could remove as no reflection in tests.

**Review by petersear — CHANGES_REQUESTED** [review]

File: aws-java-sdk-core/pom.xml (hunk 1 above)
Comment (JohnLBergqvist) [reply aws-java-sdk-core/pom.xml]: Fixed - This was done on an older release, and I was worried the change to 1.8.0 wasn't backwards compatible, but in this case it's fine.

File: aws-java-sdk-core/src/test/java/com/amazonaws/http/HttpMethodNameTest.java (hunk 2 above)
Comment (JohnLBergqvist) [reply aws-java-sdk-core/src/test/java/com/amazonaws/http/HttpMethodNameTest.java]: Removed.

File: aws-java-sdk-core/src/test/java/com/amazonaws/http/HttpMethodNameTest.java (hunk 3 above)
Comment (JohnLBergqvist) [reply aws-java-sdk-core/src/test/java/com/amazonaws/http/HttpMethodNameTest.java]: Removed.

**Review by petersear — APPROVED** [review]

**Review by sudo-jaa — APPROVED** [review]

**Review by diffblue-assistant — DISMISSED** [review]

**Review by paulkennethkent — APPROVED** [review]

**Pull request merged**